test/e2e/route_advertisements.go

```diff
@@ -222,3 +226,246 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
+				_, err := e2ekubectl.RunKubectl("", "delete", "clusteruserdefinednetwork", testCudnName, "--wait", fmt.Sprintf("--timeout=%ds", 120))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			})
+
+			ginkgo.By("Creating client pod on the udn namespace")
+			podConfig := *podConfig(echoClientPodName)
+			podConfig.namespace = f.Namespace.Name
+			clientPod = runUDNPod(cs, f.Namespace.Name, podConfig, nil)
+
+			ginkgo.By("asserting the pod UDN interface on the network-status annotation")
+			udnNetStat, err := podNetworkStatus(clientPod, func(status nadapi.NetworkStatus) bool {
+				return status.Default
+			})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			const (
+				expectedDefaultNetStatusLen = 1
+				ovnUDNInterface             = "ovn-udn1"
+			)
+			gomega.Expect(udnNetStat).To(gomega.HaveLen(expectedDefaultNetStatusLen))
+			gomega.Expect(udnNetStat[0].Interface).To(gomega.Equal(ovnUDNInterface))
+
+			gomega.Expect(len(serverContainerIPs)).To(gomega.BeNumerically(">", 0))
+		})
+		ginkgo.AfterEach(func() {
+			e2ekubectl.RunKubectlOrDie("", "delete", "ra", testRAName, "--ignore-not-found=true")
+		})
+		// -----------------               ------------------                         ---------------------
+		// |               | 172.26.0.0/16 |                |       172.18.0.0/16     | ovn-control-plane |
+		// |   external    |<------------- |   FRR router   |<------ KIND cluster --  ---------------------
+		// |    server     |               |                |                         |    ovn-worker     |   (client UDN pod advertised
+		// -----------------               ------------------                         ---------------------    using RouteAdvertisements
+		//                                                                            |    ovn-worker2    |    from default pod network)
+		//                                                                            ---------------------
+		// The client pod inside the KIND cluster on the default network exposed using default network Router
+		// Advertisement will curl the external server container sitting outside the cluster via a FRR router
+		// This test ensures the north-south connectivity is happening through podIP
+		ginkgo.It("tests are run towards the external agnhost echo server", func() {
+			ginkgo.By("routes from external bgp server are imported by nodes in the cluster")
+			externalServerV4CIDR, externalServerV6CIDR := getContainerNetworkCIDRs(bgpExternalNetworkName)
+			framework.Logf("the network cidrs to be imported are v4=%s and v6=%s", externalServerV4CIDR, externalServerV6CIDR)
+			for _, node := range nodes.Items {
+				ipVer := ""
+				cmd := []string{containerRuntime, "exec", node.Name}
+				bgpRouteCommand := strings.Split(fmt.Sprintf("ip%s route show %s", ipVer, externalServerV4CIDR), " ")
```

Comment:
should this route only be there if v4 is enabled for the cluster or do we import routes with no regards to the networks ip family? you assume v4 route is always there.

Reply:
yeah that's cause v6 single stack isn't support today with BGP... if we do add that support we must add a lane for that and also change this